### PR TITLE
Feature: Split object and tool path to separate objects to support multiple tool paths

### PR DESCRIPTION
**Pull request by zjiefee**

### CODE REVIEW

File: src/app/components/SMCanvas/Controls.jsx [hunk 1]
```diff
@@ -349,20 +351,31 @@ class Controls extends EventEmitter {
         // document.removeEventListener('mouseup', this.onDocumentMouseUp, false);
     };
 
+    disableClick() {
+        console.log('disableClick');
```
Comment (parachvte) [inline src/app/components/SMCanvas/Controls.jsx:355]: Remove `console.log()`

File: src/app/components/SMCanvas/TransformControls2D.js [hunk 2]
```diff
@@ -222,14 +220,34 @@ class TransformControls2D extends Object3D {
         */
     }
 
-    attach(object) {
-        this.object = object;
+    attach(object, selectEvent) {
```
Comment (parachvte) [inline src/app/components/SMCanvas/TransformControls2D.js:223]: What's the meaning of attach and detach?

File: src/app/widgets/CncLaserList/index.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        page: PropTypes.string.isRequired
+    };
+
+    render() {
+        const { page } = this.props;
+
+        return (
+            <div>
+                <div>
+                    {page === PAGE_EDITOR && <ObjectList {...this.props} />}
```
Comment (meiliang1123) [inline src/app/widgets/CncLaserList/index.jsx:19]: Suggestion: extends widget to hide itself on some situation

File: src/app/toolpaths/ToolPath.js [hunk 4]
```diff
@@ -98,18 +99,15 @@ class ToolPath {
...
+        this.checkoutToolPathStatus();
     }
 
     _getToolPathFiles() {
         return this.modelIDs.map(v => this.modelMap.get(v).toolPathFile);
     }
 
-    _getSelectModels() {
+    _getModels() {
+        console.log('this.modelGroup', this.modelGroup);
```
Comment (parachvte) [inline src/app/toolpaths/ToolPath.js:110]: log

**Pull request merged**